src/pagination/PaginationButton.ts

```diff
@@ -0,0 +1,106 @@
+import { callAllHandlers, isNumber } from "@chakra-ui/utils";
+import React from "react";
+import { createComponent, createHook } from "reakit-system";
+import { ButtonHTMLProps, ButtonOptions, useButton } from "reakit";
+
+import { PAGINATION_ITEM_KEYS } from "./__keys";
+import { PaginationStateReturn } from "./PaginationState";
+
+export type PaginationButtonOptions = ButtonOptions &
+  Pick<
+    PaginationStateReturn,
+    | "currentPage"
+    | "move"
+    | "next"
+    | "prev"
+    | "first"
+    | "last"
+    | "isAtMax"
+    | "isAtMin"
+  > & {
+    goto: string | number;
+    getAriaLabel?: (goto: string | number, isCurrent: boolean) => string;
+  };
```

Comment:
Let's add better TS support for the goto prop

```js

export type TGoto = "next" | "prev" | "last" | "first" | number;

export type PaginationButtonOptions = ButtonOptions &
  Pick<
    PaginationStateReturn,
    | "currentPage"
    | "move"
    | "next"
    | "prev"
    | "first"
    | "last"
    | "isAtMax"
    | "isAtMin"
  > & {
    goto: TGoto;
    getAriaLabel?: (goto: TGoto, isCurrent: boolean) => string;
  };
```

Reply:
Totally, more accurate, thanks 😄